agent/proxycfg/connect_proxy.go

```diff
@@ -231,6 +247,17 @@ func (s *handlerConnectProxy) handleUpdate(ctx context.Context, u UpdateEvent, s
 			return fmt.Errorf("invalid type for response: %T", u.Result)
 		}
 		snap.Roots = roots
+
+	case strings.HasPrefix(u.CorrelationID, peerTrustBundleWatchID):
```

Comment:
Just a general comment: would be nice if we could change "prefix-y" names to something like `peerTrustBundleWatchIDPrefix` since it would hint to a developer that you need to add something to it (otherwise if I was mucking around with watches I might not know at a glance that `peerTrustBundleWatchID` isn't a standalone correlationID).